### PR TITLE
Add httputil.DumpRequestOut & httputil.DumpResponse to SendPayload

### PR DESCRIPTION
**Pull request by xtda**
# Description

Adds DumpRequestout & DumpResponse to SendPayload for increased debugging of HTTP request & responses 

A new parameter "httpDebugging" has been added to the SendPayload method to enable/disable output

A new config entry for exchanges has been added also called httpDebugging to enable/disable per exchange

Fixes # n/a

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Usual automated testing to confirm responses are correct
Manual tested with enabled/disabled exchanges

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool 
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/coinbasepro/coinbasepro.go [hunk 1]
```diff
@@ -141,7 +143,7 @@ func (c *CoinbasePro) Setup(exch *config.ExchangeConfig) {
 	}
 }
 
-// GetProducts returns supported currency pairs on the exchange with specific
+// GetProducts returns supported currency pairs on t                                                                                                                                                                                                                                                                                                                               e exchange with specific
```
Comment (shazbert) [inline exchanges/coinbasepro/coinbasepro.go]: Probably revert this change

File: exchanges/huobihadax/huobihadax.go [hunk 2]
```diff
@@ -853,7 +855,8 @@ func (h *HUOBIHADAX) SendAuthenticatedHTTPPostRequest(method, endpoint, postBody
 	signatureParams.Set("Signature", common.Base64Encode(hmac))
 	urlPath := common.EncodeURLValues(fmt.Sprintf("%s%s", h.APIUrl, endpoint),
 		signatureParams)
-	return h.SendPayload(method, urlPath, headers, bytes.NewBufferString(postBodyValues), result, true, false, h.Verbose)
+	return h.SendPayload(method, urlPath, headers, bytes.NewBufferString(postBodyValues), result, true, false, h.Verbose,
+		false)
```
Comment (shazbert) [inline exchanges/huobihadax/huobihadax.go]: Why is this false and not h.HTTPDebugging?

File: exchanges/itbit/itbit.go [hunk 3]
```diff
@@ -403,7 +405,7 @@ func (i *ItBit) SendAuthenticatedHTTPRequest(method, path string, params map[str
 		RequestID   string `json:"requestId"`
 	}{}
 
-	err = i.SendPayload(method, urlPath, headers, bytes.NewBuffer(PayloadJSON), &intermediary, true, true, i.Verbose)
+	err = i.SendPayload(method, urlPath, headers, bytes.NewBuffer(PayloadJSON), &intermediary, true, true, i.Verbose, false)
```
Comment (shazbert) [inline exchanges/itbit/itbit.go]: same here

File: exchanges/request/request.go [hunk 4]
```diff
@@ -422,9 +432,17 @@ func (r *Requester) SendPayload(method, path string, headers map[string]string,
 		return err
 	}
 
+	if httpDebugging {
```
Comment (shazbert) [inline exchanges/request/request.go]: Could method this out to cut down on code.

File: exchanges/huobihadax/huobihadax.go (hunk 2 above)
Comment (xtda) [reply exchanges/huobihadax/huobihadax.go]: Missed that one !

File: exchanges/alphapoint/alphapoint.go [hunk 5]
```diff
@@ -51,6 +51,7 @@ type Alphapoint struct {
 // SetDefaults sets current default settings
 func (a *Alphapoint) SetDefaults() {
 	a.APIUrl = alphapointDefaultAPIURL
+	a.HTTPDebugging = false
```
Comment (thrasher-) [inline exchanges/alphapoint/alphapoint.go]: This will be false by default

File: exchanges/bitstamp/bitstamp.go [hunk 6]
```diff
@@ -645,7 +647,7 @@ func (b *Bitstamp) TransferAccountBalance(amount float64, currency, subAccount s
 
 // SendHTTPRequest sends an unauthenticated HTTP request
 func (b *Bitstamp) SendHTTPRequest(path string, result interface{}) error {
-	return b.SendPayload(http.MethodGet, path, nil, nil, result, false, false, b.Verbose)
+	return b.SendPayload(http.MethodGet, path, nil, nil, result, false, false, b.Verbose, false)
```
Comment (thrasher-) [inline exchanges/bitstamp/bitstamp.go]: Should be b.HTTPDebugging

File: exchanges/request/request.go [hunk 7]
```diff
@@ -337,6 +339,14 @@ func (r *Requester) DoRequest(req *http.Request, path string, body io.Reader, re
...
+			dump, err := httputil.DumpResponse(resp, true)
+			if err != nil {
+				log.Errorf("DumpResponse invalid response: %v:", err)
+			}
+			log.Debugf("DumpResponse: %q", dump)
+		}
+
 		resp.Body.Close()
 		if verbose {
 			log.Debugf("HTTP status: %s, Code: %v", resp.Status, resp.StatusCode)
```
Comment (gloriousCode) [inline exchanges/request/request.go]: Could we remove this line now that we have such a util?
https://github.com/thrasher-/gocryptotrader/pull/298/files#diff-d0bee293ec050b1dd3e914acdef7f6a6R353
(I cant add a comment to non affected code)

File: exchanges/alphapoint/alphapoint.go (hunk 5 above)
Comment (xtda) [reply exchanges/alphapoint/alphapoint.go]: Thanks!
I have removed the setting of default value for HTTPDebugging across all exchanges

File: exchanges/request/request.go (hunk 7 above)
Comment (xtda) [reply exchanges/request/request.go]: Great suggestion!

After discussion DumpResponse has been changed to not include the body to stop the double up

File: exchanges/request/request.go [hunk 8]
```diff
@@ -340,7 +340,7 @@ func (r *Requester) DoRequest(req *http.Request, path string, body io.Reader, re
 		}
 
 		if httpDebug {
-			dump, err := httputil.DumpResponse(resp, true)
+			dump, err := httputil.DumpResponse(resp, false)
```
Comment (gloriousCode) [inline exchanges/request/request.go]: If I had HTTPDebugging set to true and verbose false, then I wouldn't receive the response. I think it would be better still to set this to true and remove the verbose log instead.
